TableTennisGenerator/TableTennisScorer/Scorer.cs

```diff
@@ -1,11 +1,117 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Text;
 
 namespace TableTennisScorer
 {
     public class Scorer
     {
+        private const int FIRST_NAME = 2;
+        private const int FIRST_PARTNER_NAME = 3;
+        private const int SECOND_NAME = 4;
+        private const int SECOND_PARTNER_NAME = 5;
+        private const int FIRST_SCORE = 6;
+        private const int SECOND_SCORE = 7;
 
+        private const string GamesPlayed = "GamesPlayed";
+        private const string GamesWon = "GamesWon";
+        private const string PointsScored = "PointsScored";
+        private const string PointsLost = "PointsLost";
+
+        private const string PointsPerGame = "PointsPerGame";
+        private const string PointDifferential = "PointDifferential";
+
+        private string _filePath;
+        private Dictionary<string, Dictionary<string, double>> _playerMetrics;
+        public Scorer(string filePath)
+        {
+            _filePath = filePath;
+            _playerMetrics = new Dictionary<string, Dictionary<string, double>>();
+        }
+
+        public Dictionary<string, Dictionary<string, double>> GenerateMetrics()
+        {
+            using (StreamReader reader = new StreamReader(_filePath))
+            {
+                while (!reader.EndOfStream)
+                {
+                    string line = reader.ReadLine();
+                    string[] parts = line.Split(",");
+
+                    if (parts.Length == 8 && int.TryParse(parts[FIRST_SCORE], out int firstScore) && int.TryParse(parts[SECOND_SCORE], out int secondScore))
+                    {
+                        if (!_playerMetrics.ContainsKey(parts[FIRST_NAME]))

```

Comment:
what if instead of 4 if blocks we loop over [FIRST_NAME, FIRST_PARTNER_NAME, SECOND_NAME, SECOND_PARTNER_NAME] since we're doing the same thing for each? I think that might be a bit more readable